### PR TITLE
fix woodpecker_server to work out of the box

### LINKED ISSUE #74: agent defaults fail: "woodpecker-server.<namespace>.svc.cluster.local: no such host"
rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp: lookup woodpecker-server.<namespace>.svc.cluster.local: no such host"

### PR DESCRIPTION
**Pull request by ashtonian**
fixes #74

### CODE REVIEW

**Review by pat-s — COMMENTED** [review]
I am not sure this will work.
Did you try this? Usually you need to use this notation for the service endpoint.

We could make use of `WOODPECKER_BACKEND_K8S_NAMESPACE` and only require this but I am not sure if the env var would be properly expanded?

**Comment by ashtonian** [comment]
Thanks for looking at it. It works if they are in the same namespace, https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/, within the namespace its the same as a hostname lookup where the service 'woodpecker-server' is the hostname and the domain '.<namespace>.svc.cluster.local' gets appended automatically with each dns query. You therefore only need the fqdn when going outside the namespace.

File: values.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ agent:
 
   env:
     # -- Add the environment variables for the agent component
-    WOODPECKER_SERVER: "woodpecker-server.<namespace>.svc.cluster.local:9000"
+    WOODPECKER_SERVER: "woodpecker-server:9000"
```
Comment (pat-s) [inline values.yaml:29]:
```suggestion
    WOODPECKER_SERVER: "woodpecker-server:9000"
```

**Comment by pat-s** [comment]
Makes sense, thanks for the explanation.

We had split charts for a long time and user were often deploying into separate namespaces therefore. But now with the umbrella chart the namespace is the same, so this change could remove the explicit definition by users.

So yeah I see how this change would simplify things. Any objections @anbraten?

**Comment by anbraten** [comment]
Sounds fine to me

**Review by pat-s — APPROVED** [review]

**Comment by ashtonian** [comment]
🎉  Thanks!

**Pull request merged**